eventually-v0-5-0/src/metadata.rs

```diff
@@ -0,0 +1,82 @@
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum Value {
+    String(String),
+    Number(f64),
+    Boolean(bool),
+}
+
+#[derive(Debug, Clone, Default, PartialEq, Serialize, Deserialize)]
+pub struct Metadata(HashMap<String, Value>);
```

Comment:
what is the `Metadata` used for?

Reply:
Maybe I should add the documentation comments in this PR 😅 

Well, generally speaking metadata is for data that is not relevant to the domain (and thus should not be part of the Domain Event). Example: timestamp of when the Event has been recorded in the code, timestamp of when the Event has been recorded in the database, IP address of the client requesting a specific command that caused a Domain Event, etc.

Does it make sense? 👀

Reply:
i have a few questions

- did the old implementation have this?
- do you have a specific use-case in mind for this *today*?

i guess what i'm asking is, is this required now, or is this being added speculatively?

Reply:
No, old implementation didn't have it because I didn't know how to represent it.

Yes, it would be used from new version (e.g. timestamps of when events are recorded in the system).